packages/web-components/fast-foundation/src/tooltip/tooltip.ts

```diff
@@ -307,43 +318,98 @@ export class Tooltip extends FoundationElement {
         );
     };
 
+    /**
+     * mouse enters region
+     */
+    private handleRegionMouseOver = (ev: Event): void => {
+        this.isRegionHovered = true;
+    };
+
+    /**
+     * mouse leaves region
+     */
+    private handleRegionMouseOut = (ev: Event): void => {
+        this.isRegionHovered = false;
+        this.startHideDelayTimer();
+    };
+
     /**
      * mouse enters anchor
      */
     private handleAnchorMouseOver = (ev: Event): void => {
-        this.startHoverTimer();
+        if (this.tooltipVisible) {
+            // tooltip is already visible, just set the anchor hover flag
+            this.isAnchorHoveredFocused = true;
+            return;
+        }
+        this.startShowDelayTimer();
     };
 
     /**
      * mouse leaves anchor
      */
     private handleAnchorMouseOut = (ev: Event): void => {
         this.isAnchorHoveredFocused = false;
-        this.updateTooltipVisibility();
-        this.clearDelayTimer();
+        this.clearShowDelayTimer();
+        this.startHideDelayTimer();
     };
 
+    /**
+     * anchor gets focus
+     */
     private handleAnchorFocusIn = (ev: Event): void => {
-        this.startHoverTimer();
+        this.startShowDelayTimer();
     };
 
+    /**
+     * anchor loses focus
+     */
     private handleAnchorFocusOut = (ev: Event): void => {
         this.isAnchorHoveredFocused = false;
-        this.updateTooltipVisibility();
-        this.clearDelayTimer();
+        this.clearShowDelayTimer();
+        this.startHideDelayTimer();
+    };
+
+    /**
+     * starts the hide timer
+     */
+    private startHideDelayTimer = (): void => {
+        if (!this.tooltipVisible) {
+            return;
+        }
+
+        if (this.hideDelayTimer !== null) {
```

Comment:
nit pick: `this.hideDelayTimer !== null` check isn't needed. It happens in `this.clearHideDelayTimer()`.

Also, it might be better to have this before the check if tooltip isn't visible on line 377 to make sure that it's always cleared.

Reply:
done